comms/dht/src/store_forward/saf_handler/task.rs

```diff
@@ -375,15 +376,18 @@ where S: Service<DecryptedDhtMessage, Response = (), Error = PipelineError>
                 },
             })
             .map(|_| ())
-            .ok_or(StoreAndForwardError::InvalidDestination)
+            .ok_or_else(|| StoreAndForwardError::InvalidDestination)
     }
 
     fn check_signature(origin: &DhtMessageOrigin, body: &[u8]) -> Result<(), StoreAndForwardError> {
         signature::verify(&origin.public_key, &origin.signature, body)
             .map_err(|_| StoreAndForwardError::InvalidSignature)
-            .and_then(|is_valid| match is_valid {
-                true => Ok(()),
-                false => Err(StoreAndForwardError::InvalidSignature),
+            .and_then(|is_valid| {
+                if is_valid {
+                    Ok(())
+                } else {
+                    Err(StoreAndForwardError::InvalidSignature)
+                }
```

Comment:
Left looks more readable imho

Reply:
I think this has to do with performance, if is faster than a match

Reply:
I dunno. Seems like something the compiler would be smart enough to figure out

Reply:
Pure style thing, and /i simply disagree with Clippy here :)

https://rust-lang.github.io/rust-clippy/master/#match_bool

Reply:
I'm not asking to change this back BTW, just a comment